six/two/two.cpp

```diff
@@ -341,7 +341,6 @@ bool Two::getCheck(SixPyObject *py_class, const char *init_config_str, const cha
     Py_XDECREF(name);
     Py_XDECREF(check_id);
     Py_XDECREF(init_config);
-    Py_XDECREF(instance);
```

Comment:
This deserves one or more comments on why we intentionally don't decref `instance`.

I'm thinking:
* one comment at the beginning of the `done` block
* one comment when we use `PyTuple_SetItem` to say explicitly that the reference is stolen

Reply:
I agree we should definitely better document this. 

Wondering if it wouldn't be better to use `PyObject_SetItem()` instead for a more consistent developer experience in the future? Although maybe having this explicitly documented is less likely to re-introduce this tricky bug in the future. If we do change this to use `PyObject_SetItem()`, we should add a big fat comment too. 

I'm fine with merging this as-is once the comment is in place.

Reply:
@arbll go ahead and merge it once you add the nice big fat comments. If we decide to refactor some of this we can do it in a later PR.